nmostesting/mocks/Registry.py

```diff
@@ -271,3 +427,204 @@ def heartbeat(version, node_id):
+                    if len(base_data) < limit:
+                        base_data.append(value)
+                    else:
+                        new_until = value['version']
+                        break
+
+    response = Response(json.dumps(base_data), mimetype='application/json')
+
+    # add pagination headers for v1.1 and up
+    if NMOSUtils.compare_api_version("v1.1", version) <= 0:
+        protocol = "http"
+        host = get_default_ip()
+        port = str(registry.get_data().port)
+        if ENABLE_HTTPS:
+            protocol = "https"
+
+        # Always add a next for any resources that are added/modified after "now"
+        link = "<" + protocol + "://" + host + ":" + port \
+            + "/x-nmos/query/" + version + "/" + resource_type + "s/?paging.since=" + new_until \
+            + "&paging.limit=" + str(registry.paging_limit) + ">; rel=\"next\""
+
+        if since != MIN_SINCE:
+            link += ",<" + protocol + "://" + host + ":" + port \
+                + "/x-nmos/query/" + version + "/" + resource_type + "s/?paging.since=0:0&paging.limit=" \
+                + str(registry.paging_limit) + ">; rel=\"first\""
+
+        response.headers["Link"] = link
+        response.headers["X-Paging-Limit"] = registry.paging_limit
+        response.headers["X-Paging-Since"] = since
+        response.headers["X-Paging-Until"] = new_until
+
+    return response
+
+
+@REGISTRY_API.route('/x-nmos/query/<version>/<resource>/<resource_id>', methods=['GET'], strict_slashes=False)
+def get_resource(version, resource, resource_id):
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    registry.query_api_called = True
+
+    resource_type = resource.rstrip("s")
+    data = []
+    try:
+        data = registry.get_resources()[resource_type][resource_id]
+    except Exception:
+        abort(404)
+        pass
+
+    return Response(json.dumps(data), mimetype='application/json')
+
+
+@REGISTRY_API.route('/x-nmos/query/<version>/subscriptions', methods=["POST"])
+def post_subscription(version):
+
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    subscription_request = request.json
+
+    subscription_response = {}
+    created = False
+
+    try:
+        # Note: 'secure' not required in request, but is required in response
+        secure = subscription_request['secure'] if 'secure' in subscription_request else ENABLE_HTTPS
+
+        # Validate that this mock can satisfy the subscription request
+        if secure or subscription_request['max_update_rate_ms'] != 100 or len(subscription_request['params']) > 0:
```

Comment:
That now seems like it might be overly restrictive? Why not accept any value that is "sufficiently small" that they probably won't be able to tell that you're ignoring it, and then insist that subsequent requests are the same, since I think you're now storing the actual requested object in order to return it from the `/subscriptions` endpoint?

Reply:
Needs further discussion regarding how permissive/restrictive the subscription requests need to be for the purposes of testing